### ISSUE TITLE
Configuration tables should take regular objects

### ISSUE THREAD

**Issue by carlo-quinonez** [ptc-iot-sharing/ThingworxVSCodeProject#25]
Currently, configuration tables require a class definition.

I see this [line](https://github.com/ptc-iot-sharing/ThingworxVSCodeProject/blob/f9d2c5b071815759f657b3bc4cdfaa8b0e0b8f52/static/base/Decorators.d.ts#L16) is where we use the `new` keyword with the configuration table parameter, which explains why the parameter has to be a class.

```ts
declare function ConfigurationTables(tables: new (...args) => any): <T extends new (...args) => GenericThing>;(target: T) => void;
```

I'm still coming up to speed with advanced TS types, but is there a way to to use `Object.create(arg)` instead of `new (...args)`?

Also, the logic for handling configuration tables is also over [here](https://github.com/BogdanMihaiciuc/ThingTransformer/blob/024d85c5a6b1847a181b7d87cd83107aea472970/src/transformer/ThingTransformer.ts#L1670) in the transformer repo. Do you think it's worth considering moving code for building the entities into a single repo?

**Comment by BogdanMihaiciuc** [comment]
The reason for this is that regular object literals don't support the same kind of syntax that classes do. For example, you can't apply decorators to object literal properties and you also can't have properties without specifying a value for them so I settled on using an anonymous class in this case.

**Comment by carlo-quinonez** [comment]
The reason I asked about this was because of the eslint rule `max-classes-per-file`. Obviously we want to enforce 1 Class per file, but the anonymous class counts and throws an error in eslint. I've disabled the rule for now.

I found an [old eslint issue](max-classes-per-file) to ignore ClassExpressions (ie. anonymous classes). I'm following up on the [eslint repo](https://github.com/eslint/eslint/discussions/14684) about this...
